File: src/QuizDev.Core/Entities/Question.cs

namespace QuizDev.Core.Entities;

public class Question
{
    public Guid Id { get; set; }
    public string Text { get; set; }
    public Guid QuizId { get; set; }
    public Quiz Quiz { get; set; }
    public List<AnswerOption> Options { get; set; }
    public int Order { get; set; }
}

File: frontend/src/app/interfaces/Question.ts
import { AnswerOption } from "./AnswerOption";

export interface Question {
  id: string,
  text: string,
  quizId: string,
  order: number,
  options: AnswerOption[] | null
}

export interface CreateQuestion {
  text: string,
  quizId: string,
  order: number,
  options: { response: string, isCorrectOption: boolean }[]
}

export const createQuestionDefault: CreateQuestion = {
  options: [
    { response: "", isCorrectOption: false },
    { response: "", isCorrectOption: false },
    { response: "", isCorrectOption: false },
    { response: "", isCorrectOption: false }
  ],
  order: 0,
  quizId: '',
  text: ''
}
